broker/hcaxlsbroker.py

```diff
@@ -413,15 +413,30 @@ def _process(self, pathToSpreadsheet, submissionUrl, projectUuid):
                 raise ValueError('Sample of type donor must have an id attribute')
             sample_id = donor["sample_id"]
 
-            if "is_living" in donor["donor"]:
-                if donor["donor"]["is_living"].lower()in ["true", "yes"]:
-                    donor["donor"]["is_living"] = True
-                elif donor["donor"]["is_living"].lower() in ["false", "no"]:
-                    donor["donor"]["is_living"] = False
+            if "donor" not in donor:
+                # Returns ValueError if there are no other donor.fields and donor.is_living is missing
+                raise ValueError('Field is_living for sample ' + sample_id + ' is a required field and must either contain one of yes, true, no, or false')
             else:
-                raise ValueError('Field is_living in sample ' + sample_id + ' must either contain one of yes, true, no or false')
+                if "is_living" in donor["donor"]:
+                    if donor["donor"]["is_living"].lower()in ["true", "yes"]:
+                        donor["donor"]["is_living"] = True
+                    elif donor["donor"]["is_living"].lower() in ["false", "no"]:
+                        donor["donor"]["is_living"] = False
+                    '''
+                    # Commented out because we shouldn't be doing content validation in the converter
```

Comment:
I took this line out because we really shouldn't be doing any content validation in the converter script. This way, we can display the metadata content back to the user in the UI and specifically show them where the content validation errors are.

Reply:
Ok but make sure you add a handler for this somewhere else because if is_living is present but doesn't fall into one of those values, the entire json validation will fail.

Reply:
Right, but we want validation to fail. It needs to fail so we can tell the users that the value in is_living is wrong. There is a difference between things that should fail the converter, and things that should fail validation. For example, if a user tries to upload a fastq file instead of a spreadsheet, this should fail the converter and display the appropriate message in the upload UI screen. However, if a user uploads a spreadsheet but there are problems with the content (e.g. missing required field, string instead of a number), we want to display the content (which requires passing the converter) to the user in the UI along with validation error messages. The ultimate goal I think is to allow metadata changes within the UI itself.

If a required field is missing, will the converter still execute to completion? If so, I think this is fine, as we will handle validation errors after conversion, preferably through the UI.